Yosemite/Yosemite/Stores/CustomerStore.swift

```diff
@@ -67,16 +65,15 @@ public final class CustomerStore: Store {
         for siteID: Int64,
         keyword: String,
         onCompletion: @escaping (Result<Void, Error>) -> Void) {
-            searchRemote.searchCustomers(for: siteID, name: keyword) { result in
+            searchRemote.searchCustomers(for: siteID, name: keyword) { [weak self] result in
+                guard let self else { return }
                 switch result {
                 case .success(let customers):
-                    print("2 - Hit analytics endpoint")
-                    self.upsertSearchResults(siteID: siteID, readOnlySearchResults: customers) {
-                        print("4 - Saved SearchResults")
-                        self.mapSearchResultsToCustomerObject(for: siteID, with: customers) { _ in
-                            print("8 - Mapped SearchResults to Customer objects")
-                            onCompletion(.success(()))
-                        }
+                    self.upsertSearchCustomerResults(siteID: siteID, readOnlySearchResults: customers) {
+                        // We'll be saving the search results to Core Data. Not implemented yet.
+                    }
+                    self.mapSearchResultsToCustomerObject(for: siteID, with: customers) { customer in
+                        onCompletion(.success(()))
```

Comment:
Good questions!
1. I don't think these are needed, no. I was thinking I could upsert results after succeeding/failing for each Customer within the loop, but on one end this is outside the scope of this PR so I shouldn't bother with that yet, and it would be much better to just upsert all of them in one go, outside of the loop.
2. This was tricky to setup: I tried different approaches and the current implementation reflects my lack of deep understanding on how nested closures and completion handlers call each other, so I can follow the data flow correctly.  This is something I'll dedicate some more time to research today, I shared another [potential solution](https://github.com/woocommerce/woocommerce-ios/pull/7824/commits/783d932e3cb825e3c00ab1708cfed9c9681444a4) using DispatchGroup just for the learning experience, but here I was able to make the `searchCustomers` method more closer to what I initially wanted to accomplish and do all the work there.